hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -63,3 +68,14 @@ def test_valid_domains_arguments(max_length, max_element_length):
 @pytest.mark.parametrize("strategy", [domains(), urls()])
 def test_find_any_non_empty(strategy):
     find_any(strategy, lambda s: len(s) > 0)
+
+
+@given(_url_fragments_strategy)
+def test_url_fragments_contain_legal_chars(fragment):
+    assert fragment.startswith("#")
+
+    # Strip all legal escapes sequences. Any remaining % characters were not
+    # part of a legal escape sequence.
+    without_escapes = re.sub(r"(?ai)%[0-9a-f][0-9a-f]", "", fragment[1:])
+
+    assert all(c in FRAGMENT_SAFE_CHARACTERS for c in without_escapes)
```

Comment:
Side note: For some reason pytest's assertion-rewriter won't print subexpressions of a generator expression, which is annoying because it means that we don't get to see what `without_escapes` is when this assertion fails.

Reply:
I rewrote this assertion to use `set(without_escapes).issubset(FRAGMENT_SAFE_CHARACTERS)` instead.